File: db4oj.tests/src/com/db4o/db4ounit/common/handlers/ObjectContainerAdapterFactory.java
/* Copyright (C) 2010 Versant Inc. http://www.db4o.com */
package com.db4o.db4ounit.common.handlers;

/**
 * @exclude
 */
public class ObjectContainerAdapterFactory {
	public static final ObjectContainerAdapter PRE7_1_FACADE = new Pre7_1ObjectContainerAdapter(); 
	public static final ObjectContainerAdapter POST7_1_FACADE = new Post7_1ObjectContainerAdapter();
	
	public static ObjectContainerAdapter forVersion(int major, int minor) {
		if ( (major == 7 && minor >= 1) || major > 7 ) {
        	return POST7_1_FACADE;
        }
        else {
        	return PRE7_1_FACADE;
        }        		
	}
}

File: db4o.net/Db4objects.Db4o.Tests/Db4objects.Db4o.Tests/Common/Handlers/ObjectContainerAdapterFactory.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4objects.Db4o.Tests.Common.Handlers;

namespace Db4objects.Db4o.Tests.Common.Handlers
{
	/// <exclude></exclude>
	public class ObjectContainerAdapterFactory
	{
		public static readonly IObjectContainerAdapter Pre71Facade = new Pre7_1ObjectContainerAdapter
			();

		public static readonly IObjectContainerAdapter Post71Facade = new Post7_1ObjectContainerAdapter
			();

		public static IObjectContainerAdapter ForVersion(int major, int minor)
		{
			if ((major == 7 && minor >= 1) || major > 7)
			{
				return Post71Facade;
			}
			else
			{
				return Pre71Facade;
			}
		}
	}
}
